source/extensions/clusters/aggregate/cluster_lb.h

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/upstream/load_balancer_impl.h"
+#include "common/upstream/upstream_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Aggregate {
+
+class AggregateLoadBalancerContext : public Upstream::LoadBalancerContext {
+public:
+  AggregateLoadBalancerContext(Upstream::LoadBalancerContext* context,
+                               Upstream::LoadBalancerBase::HostAvailability host_avail,
+                               uint32_t host_priority)
+      : context_(context), host_avail_(host_avail), host_priority_(host_priority) {
+    if (context_ == nullptr) {
+      context_ = new Upstream::LoadBalancerContextBase();
+      own_context_ = true;
+    }
+  }
+
+  ~AggregateLoadBalancerContext() override {
+    if (own_context_) {
+      delete context_;
+    }
+  }
+
+  // Upstream::LoadBalancerContext
+  absl::optional<uint64_t> computeHashKey() override { return context_->computeHashKey(); }
+  const Network::Connection* downstreamConnection() const override {
+    return context_->downstreamConnection();
+  }
+  const Router::MetadataMatchCriteria* metadataMatchCriteria() override {
+    return context_->metadataMatchCriteria();
+  }
+  const Http::HeaderMap* downstreamHeaders() const override {
+    return context_->downstreamHeaders();
+  }
+  const Upstream::HealthyAndDegradedLoad&
+  determinePriorityLoad(const Upstream::PrioritySet&,
+                        const Upstream::HealthyAndDegradedLoad& original_priority_load) override {
+    priority_load_ = original_priority_load;
```

Comment:
`original_priority_load` is a const reference. You mentioned this in previous comment.

Reply:
i understand the need for `priority_load_`, but since you're resetting to a fixed value in the code below i think this assignment here is redundant, unless I'm missing something?

Reply:
Oh... I get what you mean. I will check and remove the assignment if it's unnecessary.